src/_P098_CacheReader.ino

```diff
@@ -0,0 +1,478 @@
+    return filename;
+  }
+  void nextFile(){
+    if (fileOpen){
+      cache.close();
+      fileOpen = false;
+    }
+    (*this->fileNr)++;
+    int fnr = (int)(*this->fileNr);
+    char *fname = new char[24];
+    for (int i = 0 ; i<24; i++){
+      fname[i] = '\0';
+    }
+    sprintf(fname,"cache_%d.bin",fnr);
+    if (fileExists(fname)){
+      (*this->sampleIndex) = 0;
+      (*this->fileNr) = fnr;
+      //this->setData();
+      //cache.close();
+      cache = tryOpenFile(fname,"r");
+      fileOpen = true;
+    } else {
+    (*this->fileNr) = 0;
+    (*this->sampleIndex) = 0;
+    }
+    //fnr++;
+    
+    delete[] fname;
+  }
+};
+
+struct P098_data_struct : public PluginTaskData_base {
+  Cache_t *cache = nullptr;
+  P098_data_struct() {}
+  ~P098_data_struct(){
+    reset();
+  }
+  void reset(){
+    if (cache != nullptr){
+      delete cache;
+      cache = nullptr;
+    }
+  }
+  bool init(){
+    reset();
+    cache = new Cache_t();
+    return isInitialized();
+  }
+  bool isInitialized() const {
+    return cache != nullptr;
+  }
+};
+
+boolean Plugin_098(byte function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+      {
+        Device[++deviceCount].Number = PLUGIN_ID_098;
+        Device[deviceCount].Type = DEVICE_TYPE_DUMMY;
+        Device[deviceCount].VType = SENSOR_TYPE_SINGLE;
+        Device[deviceCount].Ports = 0;
+        Device[deviceCount].PullUpOption = false;
+        Device[deviceCount].InverseLogicOption = false;
+        Device[deviceCount].FormulaOption = false;
+        Device[deviceCount].DecimalsOnly = true;
+        Device[deviceCount].ValueCount = 4;
+        Device[deviceCount].SendDataOption = true;
+        Device[deviceCount].TimerOption = true;
+        Device[deviceCount].GlobalSyncOption = true;
+        break;
+      }
+
+    case PLUGIN_GET_DEVICENAME:
+      {
+        string = F(PLUGIN_NAME_098);
+        break;
+      }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+      {
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_098));
+        break;
+      }
+
+    case PLUGIN_WEBFORM_LOAD:
+      {
+        sensorTypeHelper_webformLoad_allTypes(event, 0);
+        success = true;
+        break;
+      }
+
+    case PLUGIN_WEBFORM_SAVE:
+      {
+        sensorTypeHelper_saveSensorType(event, 0);
+        success = true;
+        break;
+      }
+
+    case PLUGIN_INIT:
+      {
+        initPluginTaskData(event->TaskIndex, new P098_data_struct());
+        P098_data_struct *P098_data =
+          static_cast<P098_data_struct *>(getPluginTaskData(event->TaskIndex));
+        P098_data->init();
+        P098_data->cache->initialize(event);
+        success = true;
+        break;
+      }
+    case PLUGIN_EXIT: {
+      clearPluginTaskData(event->TaskIndex);
+      success = true;
+      break;
+    }
+    case PLUGIN_READ:
+      {
+        success = true;
```

Comment:
Not sure if you need to set success to true here.
When set to true (in `PLUGIN_READ`) you also send out a number of events to signal new values.
But this plugin doesn't yield any new samples.